llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8652,6 +8660,15 @@ VPWidenRecipe *VPRecipeBuilder::tryToWiden(Instruction *I,
       NewOps[1] = GetConstantViaSCEV(NewOps[1]);
     }
     return new VPWidenRecipe(*I, make_range(NewOps.begin(), NewOps.end()));
+  }
+  case Instruction::ExtractValue: {
+    SmallVector<VPValue *> NewOps(Operands);
+    Type *I32Ty = IntegerType::getInt32Ty(I->getContext());
+    for (unsigned Idx : cast<ExtractValueInst>(I)->getIndices())
```

Comment:
Shouldn't there only be one index? If there is more than one index will it even work? I assume that means we have structs of structs, which we don't currently support vectorising. Perhaps you can just assert there is only one index?

Reply:
It didn't seem necessary to assert that here (as it's easy just to make the `extractvalue` node). Where it matters (e.g. `VPWidenRecipe::execute`) I do assert that there's one index.

Reply:
If we can reach here with multiple indices, where do we reject multiple indices before execute?

Reply:
We can't reach here with multiple indices (due to the limits on struct-vectorization). Writing the for-loop was just a little easier. Anyway, I've removed this and added an extra assert now :+1: